Lib/lib2to3/Grammar.txt

```diff
@@ -155,13 +156,15 @@ arglist: argument (',' argument)* [',']
 # Illegal combinations and orderings are blocked in ast.c:
 # multiple (test comp_for) arguments are blocked; keyword unpackings
 # that precede iterable unpackings are blocked; etc.
-argument: ( test [comp_for] |
+argument: ( test [comp_for_or_async] |
             test '=' test |
 	    '**' expr |
 	    star_expr )
 
-comp_iter: comp_for | comp_if
-comp_for: [ASYNC] 'for' exprlist 'in' testlist_safe [comp_iter]
+comp_for_or_async: comp_async_for | comp_for
+comp_iter: comp_async_for | comp_for | comp_if
+comp_async_for: 'async' 'for' exprlist 'in' testlist_safe [comp_iter]
+comp_for: 'for' exprlist 'in' testlist_safe [comp_iter]
 comp_if: 'if' old_test [comp_iter]
```

Comment:
Adding to the point raised for changes in `Grammar`: when you change/rename/add new productions here, you might be breaking some code that uses `lib2to3` to define custom fixters. Again, I'd avoid any changes.